src/models/reposyncstate.ts

```diff
@@ -218,6 +218,35 @@ export class RepoSyncState extends Model implements RepoSyncStateProperties {
 		});
 	}
 
+	static async findRepositoriesBySubscriptionIdAndRepoName(
+		subscriptionId: number,
+		page: number,
+		limit: number,
+		repoName?: string
+	): Promise<RepoSyncState[] | null> {
+
+		const whereClause: any = {
+			subscriptionId
+		};
+
+		if (repoName) {
+			whereClause.repoName = {
+				[Op.iLike]: `%${repoName}%`
```

Comment:
btw, should it be case insensitive?

Reply:
It is. In the test "Should return all matching repos across multiple orgs when repoName is provided but workspaceId is no" `new` is passed as a query param in lower case. The response, defined in `createReposWhenRepoNameIsPassedAsParam`, returns 3 repos across 2 separate orgs: `new-repo`, `another-new-rep` and `imNew`